src/aws/nodemanager/nodemanager.py

```diff
@@ -92,6 +97,27 @@ def add_task(self, task):
         """
         raise NotImplementedError()
 
+    def steal_task(self):
+        """Steals a task from the worker with the most number of tasks"""
+        assignments = []
+        for worker in self._task_assignment.keys():
+            assignments.append(len(self._task_assignment[worker]))
+        victim_worker = list(self._task_assignment.keys())[assignments.index(max(assignments))]
+        stolen_task = self._task_assignment[victim_worker].pop(0)
+        return stolen_task
+
+    def worker_change(self):
+        """
+        Based on the new running and pending workers provided by the IM. Finds stopped or 
+        newly created workers
+        """
+        total_workers = self.workers_running + self.workers_pending
+        stopped_workers = [worker for worker in self.available_workers if
```

Comment:
`self.available_workers` does not exist.

Reply:
Judging from the above, you probably should replace `self.available_workers` with `total_workers`

Reply:
I will adjust this